File: gcpe-MediaHub/gcpe-MediaHub.Server/Models/MediaContact.cs
using Microsoft.AspNetCore.Mvc.ModelBinding;
using Microsoft.CodeAnalysis;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace gcpe_MediaHub.Server.Models
{
    public class MediaContact
    {
        [Key]
        public int Id { get; set; }
        [BindRequired]
        public string? FirstName { get; set; }
        [BindRequired]
        public string? LastName { get; set; }
        [BindRequired]
        public bool IsPressGallery { get; set; }
        public string? JobTitle { get; set; }
        [BindRequired]
        public string? Email { get; set; }
        [BindRequired]
        public string? Phone { get; set; }
        public string? MobilePhone { get; set; }
        public string? CallInPhone { get; set; }
        public string? SocialMediaXURL { get; set; }
        public string? SocialMediaInstagramURL { get; set; }
        [BindRequired]
        public string? Location { get; set; }
 
        public string? OutletName { get; set; }
  
        public virtual ICollection<OutletAssociation>? Outlets { get; set; }
        public virtual ICollection<MediaRequest>? Requests { get; set; }

        public DateTime? LastActive { get; set; }

    }
}

File: gcpe-MediaHub/gcpe-mediahub.client/src/models/mediaContact.ts

import MediaOutlet from './mediaOutlet';
import MediaRequest from './mediaRequest';
import { OutletAssociation } from './OutletAssociation';
import { SocialMediaLink } from './SocialMediaLink';
import { PhoneNumber } from './PhoneNumber';

export class MediaContact {
    id: number| undefined;
    firstName: string | undefined;
    lastName: string | undefined;
    isPressGallery: boolean | undefined;
    isActive: boolean = true;
    email: string | undefined;
    jobTitle: string | undefined;
    jobTitleId: number = 0;
    mediaOutletContactRelationships: OutletAssociation[] = [];
    socialMedias: SocialMediaLink[] = [];
    personalWebsite: string | undefined;
    phoneNumbers: PhoneNumber[] = [];
    requests: MediaRequest[] = [];
    location: string | undefined;
    lastActive: Date | undefined;
}

export default MediaContact;